### ISSUE TITLE
History Push Error 

### ISSUE THREAD

**Issue by Schlesiger** [ui-router/core#45]
After upgrading ui-router-react from `0.3.0` to `0.4.0` and ui-router-core from `^1.0.0-beta.2` to `4.0.0`, I am getting the following error messages when using `router.stateService.go(statename)` *or* `router.urlRouter.otherwise(statename)`:

**Chrome**
```
Uncaught DOMException: Failed to execute 'replaceState' on 'History': A history state object with URL 'http://taba/' cannot be created in a document with origin 'http://localhost:8090' and URL 'http://localhost:8090/'.
```

**Firefox**
```
SecurityError: The operation is insecure.
```

I am unable to share a demo on GitLab due to [non-root base causing infinite redirect](https://github.com/ui-router/core/issues/44), but you can clone [the project](https://gitlab.com/wallzero/ui-router-react-digest/tree/master.) and run locally yourself. Run via `npm install && npm run build:demo && npm run start:demo:dev`.

If you go to the `localhost:8090`; the attempted redirect from `/` to `/taba` will cause the error. If you manually go to `localhost:8090/taba`, you will see the states properly render. However, selecting a tab will also trigger the error, and the URL will not change.

**Comment by Schlesiger** [comment]
If I remove the `<base>` tag from the `index.html`, everything works. However, it won't be possible to host on anything but `/`. It would seem the baseHref generated [here](https://github.com/ui-router/core/blob/master/src/vanilla/pushStateLocationService.ts) doesn't work when `<base>` tag is used.

**Comment by Schlesiger** [comment]
I am trying this on 5.0.0-pre.1 and the issue persists. Also, if I remove the `<base href="/">` tag, stateService.go will append to the current route instead of replacing. For example, when on taba, if trying to navigate to tabb, the URL will change from `localhost:8090/taba` to `localhost:8090/tabatabb` instead of `localhost:8090/tabb`.

If I return `<base href="/">`, url's are changed properly on transiton, but the errors mentioned above return.

**Comment by Schlesiger** [comment]
Seems to be resolved with `@ui-router/core@5.0.3` and `@uirouter/react@0.5.0` 😃
